lib/random.cu

```diff
@@ -45,43 +45,68 @@ namespace quda {
      @brief CUDA kernel to initialize CURAND RNG states
      @param state CURAND RNG state array
      @param seed initial seed for RNG
-     @param rng_size size of the CURAND RNG state array
+     @param size size of the CURAND RNG state array
      @param arg Metadata needed for computing multi-gpu offsets
   */
-  __global__ void kernel_random(cuRNGState *state, int seed, int rng_size, rngArg arg) {
+  __global__ void kernel_random(cuRNGState *state, unsigned long long seed, int size_cb, rngArg arg)
+  {
     int id = blockIdx.x * blockDim.x + threadIdx.x;
-    if (id < rng_size) {
+    int parity = blockIdx.y * blockDim.y + threadIdx.y;
+    if (id < size_cb) {
```

Comment:
Maybe this is better dealt with when we get to multi-rhs support, but should we make this support 4 and 5-d fields natively?

Reply:
Leave for multi-rhs support, since there are a bunch of things to be reworked.  Way outside the scope of this pull.